### ISSUE TITLE
distribute tensorflow read data problem

### ISSUE THREAD

**Issue by haibarasiao** [tensorflow/tensorflow#4171]
Hi,
when I want to build a distribute model with cifar-10  (https://tensorflow.googlesource.com/tensorflow/+/master/tensorflow/models/image/cifar10/),
it hanged at sess.run , I think it is the input queue problem. It can not read image and label from file.
But when run with no distribute, everything is OK.
Any suggestions?

in train():
images, labels = model.distorted_inputs()

in model.py:
def distorted_inputs():
  batch_size=FLAGS.batch_size)
  if not FLAGS.data_dir:
    raise ValueError('Please supply a data_dir')
  data_dir = os.path.join(FLAGS.data_dir, '')
  return model_input.distorted_inputs(data_dir=data_dir,
                                        batch_size=FLAGS.batch_size)

in model_input.py:
def distorted_inputs(data_dir, batch_size):
  filenames = [os.path.join(data_dir, '%d.bin' % i)
               for i in xrange(1, 5)]
  for f in filenames:
    if not gfile.Exists(f):
      raise ValueError('Failed to find file: ' + f)
  filename_queue = tf.train.string_input_producer(filenames)
  read_input = read_cifar10(filename_queue)
  reshaped_image = tf.cast(read_input.uint8image, tf.float32)
  reshaped_image = tf.Print(reshaped_image, [reshaped_image], 'this is float image')
  min_fraction_of_examples_in_queue = 0.4
  min_queue_examples = int(NUM_EXAMPLES_PER_EPOCH_FOR_TRAIN *
                           min_fraction_of_examples_in_queue)
  print ('Filling queue with %d CIFAR images before starting to train. '
         'This will take a few minutes.' % min_queue_examples)
  return _generate_image_and_label_batch(reshaped_image, read_input.label,
                                         min_queue_examples, batch_size,test)

def _generate_image_and_label_batch(image, label, min_queue_examples,
                                    batch_size,test):
  num_preprocess_threads = 16
  images, label_batch = tf.train.shuffle_batch(
      [image, label],
      batch_size=batch_size,
      num_threads=num_preprocess_threads,
      capacity=min_queue_examples + 3 * batch_size,
      min_after_dequeue=min_queue_examples)
  return images, tf.reshape(label_batch, [batch_size])

**Comment by haibarasiao** [comment]
I run one ps job and one work job at localhost.

**Comment by haibarasiao** [comment]
seems found the problem, it's the queue problem. Now everything is OK.
